changeset.go

```diff
@@ -231,35 +266,42 @@ func (cs *Changeset) DeleteResource(ctx context.Context, changesetNamespace, cha
 		"cs": tr.String(),
 	})
 	log.Infof("Deleting %v/%s", resourceNamespace, resource)
+
+	comp, err := cs.isOpCompleted(ctx, tr, id)
+	if comp {
+		log.Infof("%s '%s' is already deleted.", resource.Kind, resource.Name)
+		return nil
+	}
+
 	switch resource.Kind {
 	case KindDaemonSet:
-		return cs.deleteDaemonSet(ctx, tr, resourceNamespace, resource.Name, cascade)
+		return cs.deleteDaemonSet(ctx, tr, resourceNamespace, id, resource.Name, cascade)
```

Comment:
This is becoming unmanageable with the proliferation of arguments.
Let's move them to a type similar to:
```go
type deleteResource struct {
	cascade bool
	resource metav1.ObjectMeta
	operationID string
}
```